src/core/swap.c

```diff
@@ -701,9 +701,15 @@ static void swap_enter_active(Swap *s, SwapResult f) {
 static void swap_enter_dead_or_active(Swap *s, SwapResult f) {
         assert(s);
 
-        if (s->from_proc_swaps)
+        if (s->from_proc_swaps) {
+                Swap *other;
+
                 swap_enter_active(s, f);
```

Comment:
`f` can have different values here, and I'm a bit wary of unconditionally calling `job_finish_and_invalidate` with JOB_FAILED. Shouldn't it be something like
```C
 JobResult result = f == SWAP_SUCCESS ? JOB_DONE : JOB_FAILED;

...
    job_finish_and_invalidate(UNIT(other)->job, result, true, false);
```
?